lib/SILGen/SILGenLValue.cpp

```diff
@@ -2104,7 +2104,20 @@ static ManagedValue visitRecNonInOutBase(SILGenLValue &SGL, Expr *e,
     ctx = SGFContext::AllowGuaranteedPlusZero;
   }
 
-  return SGF.emitRValueAsSingleValue(e, ctx);
+  ManagedValue mv = SGF.emitRValueAsSingleValue(e, ctx);
```

Comment:
This LGTM.

Two thoughts:

1. We should run this past John/Joe/Slava, but when we talked about this via chat, we used the name "lvalue access scope". That really is what a formal evaluation scope is if you think about it and from a mapping from swift->SILGen/clarity perspective I wonder if we should just use that. It would be clearer.
2. The way you forwarded the normal cleanup here brings up a quality fear of mine: trying to push a ManagedValue with a lvalue access cleanup through a lexical scope. That seems to me to be a place where we probably want an assert. I imagine we would need to ifdef it out in non-asserts builds though. I don't think this issue can happen here since you are not performing such an action. But, I guess the bigger issue is that given a ManagedValue we erase the type of the cleanup. I could imagine bugs resulting from that.
3. In your comment you should write destroy_value, not release_value.